### PR TITLE
IPP code refactored

### PR DESCRIPTION
**Pull request by lupustr3**
IPP code path  implemented as separate static functions to simplify future work with IPP code and make it more readable.

### CODE REVIEW

**Comment by alalek** [comment]
BTW, Previous patch didn't introduce any performance regression for IPPICV code path.

**Comment by vpisarev** [comment]
@lupustr3, thanks! I see some quite big chunks of commented off code. Could you simply remove it?

**Comment by lupustr3** [comment]
There are some disabled IPP blocks which were commented out in original code for some reason. Modifying internals of IPP wrappers is out of the scope of this commit. And don't you think it is better leave them be so they could be reactivated in future? Cause some comments in these disabled blocks suggest that they were disabled temporary.

**Comment by vpisarev** [comment]
ok, I did not realize that you did not added new commented-off code, but rather moved some commented-off code to the new place. :+1:

File: modules/core/src/stat.cpp [hunk 1]
```diff
@@ -2181,115 +2216,120 @@ static bool ocl_minMaxIdx( InputArray _src, double* minVal, double* maxVal, int*
...
-            else
-            {
-                typedef IppStatus (CV_STDCALL* ippiMinMaxIndxFuncC1)(const void *, int, IppiSize, Ipp32f *, Ipp32f *, IppiPoint *, IppiPoint *);
-
-                CV_SUPPRESS_DEPRECATED_START
-                ippiMinMaxIndxFuncC1 ippFuncC1 =
-                    depth == CV_8U ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_8u_C1R :
-                    depth == CV_8S ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_8s_C1R :
-                    depth == CV_16U ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_16u_C1R :
-                #if !((defined _MSC_VER && defined _M_IX86) || defined __i386__)
```
Comment (alalek) [inline modules/core/src/stat.cpp]: This condition should be restored.

http://pullrequest.opencv.org/buildbot/builders/precommit_windows32/builds/196/steps/test_imgproc_1/logs/stdio
(this buildslave's CPU is AMD A10-6800K, so something went wrong in this case)

**Comment by lupustr3** [comment]
There is some problem with build jobs. Can they be restarted?

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**